### ISSUE TITLE
Two false positives with `types.MethodType`

### ISSUE THREAD

**Issue by patrick-kidger** [microsoft/pyright#4772]
```python
import types

def identity(method: types.MethodType):
    assert type(method) is types.MethodType  # passes at runtime
    return types.MethodType(method.__func__, method.__self__)  # error: Argument of type "_StaticFunctionType" cannot be assigned to parameter "__func" of type "(...) -> Any" in function "__init__"

class X:
    def f(self):
        pass

identity(X().f)  # error: Argument of type "() -> None" cannot be assigned to parameter "method" of type "MethodType" in function "identity"
```

Python 3.10
pyright 1.1.298 run from the CLI

**Comment by erictraut** [comment]
The types defined in the `types` module are internal types used within the Python interpreter. They should not be used in type annotations in general code unless you're writing a specialized tool that deals with these runtime types. Static type checkers ascribe no special meaning to these classes, so they are not compatible with callables.

I think what you are intending here is to use a `typing.Callable` type or a `TypeVar` that is bound to a `Callable`.

**Comment by patrick-kidger** [comment]
I am writing a tool that works specifically with method types. I am deliberately not using `Callable`.

Do I understand correctly that this is simply a wont-fix that I should `# pyright: ignore`?

**Comment by erictraut** [comment]
Yes, pyright is working as designed, consistent with the Python static type system.
